### PR TITLE
clean up migration notes

### PR DESCRIPTION
**Pull request by chriselion**
Made some changes to the migration docs, grouping the Agent reset/done notes together. Also added a note about Add/SetReward before Done.

### CODE REVIEW

File: docs/Migrating.md [hunk 1]
```diff
@@ -11,23 +11,28 @@ The versions can be found in
 ## Migrating from 0.13 to latest
 
 ### Important changes
+* Several changes were made to how agents are reset and marked as done:
+  * Calling `Done()` on the Agent will now reset it immediately and call the `AgentReset` virtual method. (This is to simplify the previous logic in which the Agent had to wait for the next `EnvironmentStep` to reset)
+  * The "Reset on Done" setting in AgentParameters was removed; this is now effectively always true. `AgentOnDone` virtual method on the Agent has been removed.
+  * Agents will always request a decision after being marked as `Done()` and will no longer wait for the next call to `RequestDecision()`.
```
Comment (vincentpierre) [inline docs/Migrating.md]: Is this true ? I don't think that is right, if it is, it might be a bug.

File: docs/Migrating.md [hunk 2]
```diff
@@ -11,23 +11,28 @@ The versions can be found in
...
+  * The `Info` field of the Agent has been made private. (Was only used internally and not meant to be modified outside of the Agent)
+  * The `GetReward()` method on the Agent has been removed. (It was being confused with `GetCumulativeReward()`)
+  * The `AgentAction` struct no longer contains a `value` field. (Value estimates were not set during inference)
+  * The `GetValueEstimate()` method on the Agent has been removed.
+  * The `UpdateValueAction()` method on the Agent has been removed.
+* The deprecated `RayPerception3D` and `RayPerception3D` classes were removed, and the `legacyHitFractionBehavior` argument was removed from `RayPerceptionSensor.PerceiveStatic()`.
 
 ### Steps to Migrate
+* If your Agent implemented `AgentOnDone` and did not have the checkbox `Reset On Done` checked in the inspector, you must call the code that was in `AgentOnDone` manually.
+* If you give your Agent a reward or penalty at the end of an episode (e.g. for reaching a goal or falling off of a platform), make sure you call `AddReward()` or `SetReward()` *before* calling `Done()`. Previous
```
Comment (vincentpierre) [inline docs/Migrating.md]: ~. Previous~

File: docs/Migrating.md (hunk 1 above)
Comment (chriselion) [reply docs/Migrating.md]: You wrote it :P  I just moved it. I think it was from the first set of changes.

File: docs/Migrating.md (hunk 2 above)
Comment (chriselion) [reply docs/Migrating.md]: oops, that meant to write "Previously, the order didn't matter."

Comment (chriselion) [reply docs/Migrating.md]: Updated

File: docs/Migrating.md (hunk 1 above)
Comment (chriselion) [reply docs/Migrating.md]: That line came from https://github.com/Unity-Technologies/ml-agents/pull/3242

**Pull request merged**